### PR TITLE
Add elapsed time ellipsis printing to Capture View

### PR DESCRIPTION
**Pull request by robbiesri**
For instrumented timers, sometimes, the function names are quite long, and require a very high zoom level in order to print out the function name + elapsed time.

Because viewing the elapsed time is important, this change adds the ability to prioritize printing the elapsed time with this format:

Function name: "LONG TEXT CUT OFF HERE"
Elapsed time: 4.0 us 
Old:   ["LONG TEXT CUT OFF H"]
New: ["LONG TEXT CU... 4.0 us"]

### CODE REVIEW

**Comment by robbiesri** [comment]
Hi @pierricgimmig, here's my first attempt at a PR for your project!

I wasn't entirely clear about some style guidelines in your code. In particular, I was curious about how you'd like to handle `static const` variables. I don't like magic numbers very much, so I tend to use this (or DEFINES) but I wasn't sure what the right naming convention was, nor was I totally sure where to put them. I just tried to grep the code for ideas, but definitely looking for feedback.

As far as coding questions:
* I stashed the length of the elapsed time string in the TextBox class. I don't totally know if it's the _best_ place, but it made more sense than Timer.
* I added a new method to TextRenderer to handle the ellipsis generation. I really couldn't figure out a good way to avoid adding another pass over a_Text. Maybe merging everything into AddTextInternal is the right thing to do, but I didn't like the idea of adding a bunch of extra code to AddTextInternal, when I thought TimeGraph::UpdatePrimitives was the only real client for this functionality.

Thanks in advance, looking forward to your feedback!

File: OrbitGl/TextBox.h [hunk 1]
```diff
@@ -67,6 +67,9 @@ class TextBox
 
     void SetTextY( float a_Y ){ m_TextY = a_Y; }
 
+    void SetElapsedTimeTextLength( const size_t& a_Length ){ m_ElapsedTimeTextLength = a_Length; }
```
Comment (pierricgimmig) [inline OrbitGl/TextBox.h]: a_Length should be passed by value as it will be passed through register.

File: OrbitGl/TextRenderer.cpp [hunk 2]
```diff
@@ -270,6 +270,85 @@ void TextRenderer::AddText( const char* a_Text
     AddTextInternal( m_Font, a_Text, ColorToVec4( a_Color ), &m_Pen, a_MaxSize, a_Z );
 }
 
+//-----------------------------------------------------------------------------
+void TextRenderer::AddTextTrailingTimePrioritized( const char* a_Text
+                                   , float a_X
+                                   , float a_Y
+                                   , float a_Z
+                                   , const Color& a_Color
+                                   , size_t timeStringLength
```
Comment (pierricgimmig) [inline OrbitGl/TextRenderer.cpp]: The argument name "timeStringLength" could be generalized.  We could use the same mechanism to prioritize trailing text that is not necessarily time.

File: OrbitGl/TextRenderer.cpp [hunk 3]
```diff
@@ -270,6 +270,85 @@ void TextRenderer::AddText( const char* a_Text
     AddTextInternal( m_Font, a_Text, ColorToVec4( a_Color ), &m_Pen, a_MaxSize, a_Z );
 }
 
+//-----------------------------------------------------------------------------
+void TextRenderer::AddTextTrailingTimePrioritized( const char* a_Text
```
Comment (pierricgimmig) [inline OrbitGl/TextRenderer.cpp]: Function name could be generalized to something not related to time.  AddTextTrailingCharsPrioritized maybe?

File: OrbitGl/TextBox.h [hunk 4]
```diff
@@ -94,8 +97,9 @@ class TextBox
     Color        m_Color;
     Timer        m_Timer;
     int          m_MainFrameCounter;
-    bool		 m_Selected;
+    bool         m_Selected;
     float        m_TextY;
+    size_t       m_ElapsedTimeTextLength;
```
Comment (pierricgimmig) [inline OrbitGl/TextBox.h]: Uninitialized variable.

**Comment by robbiesri** [comment]
> At first, the fact that the you can lose some characters as you zoom in (in favor of trailing time characters) was surprising,
Yeah, this is a really good point, I should have made a comment about it. I wasn't entirely sure what to do.  There might be a version of this where only _some_ leading characters get printed until the time can be printed, to prevent the whole 'losing function characters' effect. Maybe a future thing to do to look at the UX of this again :p

I actually was snooping around some other profilers, and the few I looked at had the current (pre-PR) behavio. So I couldn't cheat off other profilers 😢

File: OrbitGl/TextBox.h [hunk 5]
```diff
@@ -67,6 +67,9 @@ class TextBox
 
     void SetTextY( float a_Y ){ m_TextY = a_Y; }
 
+    void SetElapsedTimeTextLength( const size_t& a_Length ){ m_ElapsedTimeTextLength = a_Length; }
+    const size_t GetElapsedTimeTextLength() const { return m_ElapsedTimeTextLength; }
```
[the remark this answers is not in the document]
Comment (robbiesri) [reply OrbitGl/TextBox.h]: Yes, I am an idiot. Copy-paste foolishness.

File: OrbitGl/TextBox.h (hunk 1 above)
Comment (robbiesri) [reply OrbitGl/TextBox.h]: grrr, ya, fudged this also :(

File: OrbitGl/TextRenderer.cpp (hunk 3 above)
Comment (robbiesri) [reply OrbitGl/TextRenderer.cpp]: AddTextTrailingCharsPrioritized sounds good to me!

File: OrbitGl/TextRenderer.cpp (hunk 2 above)
Comment (robbiesri) [reply OrbitGl/TextRenderer.cpp]: Good point. How about... `a_TrailingCharsLength` ?

Comment (pierricgimmig) [reply OrbitGl/TextRenderer.cpp]: Yeah perfect!

File: OrbitGl/TextRenderer.cpp (hunk 3 above)
Comment (pierricgimmig) [reply OrbitGl/TextRenderer.cpp]: cool!

File: OrbitGl/TextBox.h (hunk 4 above)
Comment (robbiesri) [reply OrbitGl/TextBox.h]: RIP in peace 💀 

Fixed the constructors, will push updated PR in a bit

Comment (pierricgimmig) [reply OrbitGl/TextBox.h]: Great!  Thanks Rob!

**Pull request merged**